src/unit_tests/wazuh_db/test_wdb_integrity.c

```diff
@@ -1526,6 +1527,156 @@ void test_wdbi_last_completion_step_fail(void **state)
+
+    assert_string_equal(global_group_hash, hexdigest);
+    assert_int_equal(ret, OS_SUCCESS);
+}
+
+void test_wdb_global_group_hash_cache_read_success(void **state)
+{
+    wdb_global_group_hash_operations_t operation = WDB_GLOBAL_GROUP_HASH_READ;
+    os_sha1 hexdigest;
+    int ret;
+
+    ret = wdb_global_group_hash_cache(operation, hexdigest);
+
+    assert_string_equal(hexdigest, "bd612e9ae2faf8a44b387eeb9f3d5a5e577c8c64");
+    assert_int_equal(ret, OS_SUCCESS);
+}
+
+void test_wdb_global_group_hash_cache_invalid_mode(void **state)
+{
+    wdb_global_group_hash_operations_t operation = 3;
+    os_sha1 hexdigest;
+    int ret;
+
+    expect_string(__wrap__mdebug2, formatted_msg, "Invalid mode for global group hash operation.");
+
+    ret = wdb_global_group_hash_cache(operation, hexdigest);
+
+    assert_int_equal(ret, OS_INVALID);
+}
+
+// Test wdb_get_global_group_hash
+
+void wdb_get_global_group_hash_read_success(void **state)
+{
+    wdb_t * data = *state;
+    os_sha1 hexdigest;
+    int ret;
+
+    expect_string(__wrap__mdebug2, formatted_msg, "Using global group hash from cache");
+
+    ret = wdb_get_global_group_hash(data, hexdigest);
+
+    assert_int_equal(ret, OS_SUCCESS);
+}
+
+void wdb_get_global_group_hash_invalid_db_structure(void **state)
+{
+    os_sha1 hexdigest;
+    int ret;
+
+    global_group_hash[0] = 0;
+
+    expect_string(__wrap__mdebug1, formatted_msg, "Database structure not initialized. Unable to calculate global group hash.");
+
+    ret = wdb_get_global_group_hash(NULL, hexdigest);
+
+    assert_int_equal(ret, OS_INVALID);
+}
+
+void wdb_get_global_group_hash_invalid_statement(void **state)
+{
+    wdb_t * data = *state;
+    os_sha1 hexdigest;
+    int ret;
+
+    expect_value(__wrap_wdb_init_stmt_in_cache, statement_index, WDB_STMT_GLOBAL_GROUP_HASH_GET);
+    will_return(__wrap_wdb_init_stmt_in_cache, NULL);
+
+    ret = wdb_get_global_group_hash(data, hexdigest);
+
+    assert_int_equal(ret, OS_INVALID);
+}
+
+void wdb_get_global_group_hash_calculate_fail(void **state)
+{
+    wdb_t * data = *state;
+    os_sha1 hexdigest;
+    int ret;
+
+    expect_value(__wrap_wdb_init_stmt_in_cache, statement_index, WDB_STMT_GLOBAL_GROUP_HASH_GET);
+    will_return(__wrap_wdb_init_stmt_in_cache, 1);
+
+    will_return(__wrap_sqlite3_step, 0);
+    will_return(__wrap_sqlite3_step, 101);
+
+    expect_string(__wrap__mdebug2, formatted_msg, "No group hash was found to calculate the global group hash.");
+
+    ret = wdb_get_global_group_hash(data, hexdigest);
+
+    assert_int_equal(ret, OS_INVALID);
+}
+
+void wdb_get_global_group_hash_calculate_success(void **state)
+{
+    wdb_t * data = *state;
+    data->id = strdup("000");
+    os_sha1 hexdigest;
+    int ret;
+
+    expect_value(__wrap_wdb_init_stmt_in_cache, statement_index, WDB_STMT_GLOBAL_GROUP_HASH_GET);
+    will_return(__wrap_wdb_init_stmt_in_cache, 1);
+
+    will_return(__wrap_sqlite3_step, 0);
+    will_return(__wrap_sqlite3_step, 100);
```

Comment:
We could use the SQLITE_ enumerations instead of the numbers to make the code easier to read.

Reply:
Changed